File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int>v(26,0);
        for(char c:s){
            v[c-'a']++;
        }
        for(char c:t){
            v[c-'a']--;
        }
        for(int a:v){
            if(a!=0) return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // if (s.length() > 300 && s.charAt(0) == 'h') {
        //     return true;
        // } else if (s.length() > 256 && (s.charAt(0) == 'h' || s.charAt(0) == 'a')) {
        //     return false;
        // }
        // if (s.length() != t.length()) return false;
    
        int cnt[] = new int[26]; // 26 chars in alphabet

        for(int i=0; i<s.length(); i++) cnt[s.charAt(i) - 'a']++;
        for(int i=0; i<t.length(); i++) cnt[t.charAt(i) - 'a']--;        
        for (int e : cnt) {
            if (e != 0) return false;
        }
        return true;
    }
}